Source/WebKit/UIProcess/RemoteLayerTree/mac/RemoteLayerTreeDrawingAreaProxyMac.mm

```diff
@@ -52,22 +91,120 @@
     return makeUnique<RemoteScrollingCoordinatorProxyMac>(m_webPageProxy);
 }
 
+DisplayLink* RemoteLayerTreeDrawingAreaProxyMac::exisingDisplayLink()
+{
+    if (!m_displayID)
+        return nullptr;
+    
+    return process().processPool().displayLinks().displayLinkForDisplay(*m_displayID);
+}
+
+DisplayLink& RemoteLayerTreeDrawingAreaProxyMac::ensureDisplayLink()
+{
+    ASSERT(m_displayID);
+
+    auto& displayLinks = process().processPool().displayLinks();
+    auto* displayLink = displayLinks.displayLinkForDisplay(*m_displayID);
+    if (!displayLink) {
+        auto newDisplayLink = makeUnique<DisplayLink>(*m_displayID);
+        displayLink = newDisplayLink.get();
+        displayLinks.add(WTFMove(newDisplayLink));
+    }
+    return *displayLink;
+}
+
+void RemoteLayerTreeDrawingAreaProxyMac::removeObserver(std::optional<DisplayLinkObserverID>& observerID)
+{
+    if (!observerID)
+        return;
+
+    if (auto* displayLink = exisingDisplayLink())
+        displayLink->removeObserver(*m_displayLinkClient, *observerID);
+
+    observerID = { };
+}
+
 void RemoteLayerTreeDrawingAreaProxyMac::scheduleDisplayRefreshCallbacks()
 {
-    if (m_displayLinkTimer.isActive())
+    LOG_WITH_STREAM(DisplayLink, stream << "[UI ] RemoteLayerTreeDrawingAreaProxyMac " << this << " scheduleDisplayLink for display " << m_displayID << " - existing observer " << m_displayRefreshObserverID);
```

Comment:
is the extra space in `[UI]` there for a reason?

Reply:
Yes, to line up with `[Web]`